File: src/PcIfy.Server.Flutter/lib/core/constants/api_routes.dart
abstract final class ApiRoutes {
  static const authLogin = '/api/auth/login';
  static const systemHealth = '/api/system/health';
  static const systemInfo = '/api/system/info';
  static const filesRoots = '/api/files/roots';
  static const filesList = '/api/files/list';
  static const filesStream = '/api/files/stream';
  static const filesDownload = '/api/files/download';
  static const filesUpload = '/api/files/upload';
  static const filesDelete = '/api/files/delete';
  static const filesCopy = '/api/files/copy';
  static const filesMove = '/api/files/move';
  static const thumbnails = '/api/thumbnails';
  static const tokenQueryParam = 'token';

  static const systemControlStatus        = '/api/system/control/status';
  static const systemControlVolume        = '/api/system/control/volume';
  static const systemControlMute          = '/api/system/control/mute';
  static const systemControlLock          = '/api/system/control/lock';
  static const systemControlWake          = '/api/system/control/wake';
  static const systemControlClipboard     = '/api/system/control/clipboard';

  static const systemApps                 = '/api/system/apps';
  static const systemAppsLaunch           = '/api/system/apps/launch';
  static const systemAppsAdd              = '/api/system/apps/add';

  static const filesVideoInfo = '/api/files/video-info';

  static const streamingPrefixes = [
    filesStream,
    filesDownload,
    thumbnails,
  ];
}

File: src/PcIfy.Server/Constants/ApiRoutes.cs
namespace PcIfy.Server.Constants;

public static class ApiRoutes
{
    public const string AuthLogin = "/api/auth/login";

    public const string SystemHealth = "/api/system/health";
    public const string SystemInfo = "/api/system/info";

    public const string FilesRoots = "/api/files/roots";
    public const string FilesList = "/api/files/list";
    public const string FilesStream = "/api/files/stream";
    public const string FilesDownload = "/api/files/download";
    public const string FilesUpload = "/api/files/upload";
    public const string FilesDelete = "/api/files/delete";
    public const string FilesCopy = "/api/files/copy";
    public const string FilesMove = "/api/files/move";

    public const string Thumbnails = "/api/thumbnails";

    public const string TokenQueryParam = "token";

    public const string SystemControlStatus        = "/api/system/control/status";
    public const string SystemControlVolume        = "/api/system/control/volume";
    public const string SystemControlMute          = "/api/system/control/mute";
    public const string SystemControlLock          = "/api/system/control/lock";
    public const string SystemControlWake          = "/api/system/control/wake";
}
